### ISSUE TITLE
Not working when adding methods to Array.prototype

### ISSUE THREAD

**Issue by pascalvaccaro** [BohdanTkachenko/webpack-split-by-path#23]
Hi there,
Curiously, while my application was running great when launching webpack server from source files, I could not get it to build with webpack-split. I got this error: 

```
D:\dev\stime\projects\wks\d-marketing\d-marketing-ui\node_modules\webpack-split-by-path\index.js:70
      if (ignore[i].test(userRequest)) {
                    ^
TypeError: ignore[i].test is not a function
```

While debugging, I noticed 1st that the 'ignore' array was empty. Then, I also noticed that ignore[i] was actually returning a method. 
Reason was I overloaded the Array.prototype object with a custom method I use a lot in my code (more convenient this way, imo), but the loop being used (`for ... in`) loops over the object, not the array. So the inject method is called, has no 'test' method and it bugs (=> no build!!!)

Is it possible to change this loop into: 

```
for (var i = 0; i < ignore.length; ++i) {
      if (ignore[i].test(userRequest)) {
        return match;
      }
    }
```

So that we can overload `Array.prototype` without error when building with webpack?

**Comment by pascalvaccaro** [comment]
Also, you could change the original loop but adding a condition:

```
if (ignore.length) {
  for (var i in ignore) {
     if (ignore[i].test(userRequest)) {
       return match;
     }
   }
}
```

**Comment by BohdanTkachenko** [comment]
Hi @Ripley511 

We just released 2.0.0 version of this plugin for webpack 2. 
For webpack 1 there is still a branch webpack1 and version 1.0.0 on NPM.
Hi @aldendaniels 

Please consider upgrading to webpack 2 and version 2.0.0 of this plugin.
If this issue still reproduces on 2.0.0 please re-open this issue.
If you really need to have this in 1.0.0, PR's to webpack1 branch are always welcome, but I don't longer plan to fix issues for webpack1.

Thanks!
